### ISSUE TITLE
can't server index at root with wai-app-static

### ISSUE THREAD

**Issue by winterland1989** [yesodweb/wai#376]
i want to server a static webpage using WaiAppStatic.Storage.Embedded like this:

``` haskell
{-# LANGUAGE TemplateHaskell, QuasiQuotes, OverloadedStrings #-}

module Static (mkEmbedded) where

import WaiAppStatic.Storage.Embedded
import Crypto.Hash.MD5 (hashlazy)
import Network.Mime (MimeType)
import qualified Data.ByteString.Lazy as BL
import qualified Data.ByteString.Base64 as B64
import qualified Data.Text as T
import qualified Data.Text.Encoding as T
import System.FilePath ((</>))

hash :: BL.ByteString -> T.Text
hash = T.take 8 . T.decodeUtf8 . B64.encode . hashlazy

staticFiles :: [(FilePath, MimeType)]
staticFiles = [
        ("index.html"         , "text/html")
    ,   ("main.js"            , "application/x-javascript")
    ,   ("html-sanitizer.js"  , "application/x-javascript")
    ,   ("ext.js"             , "application/x-javascript")
    ,   ("main.css"           , "text/css")
    ]

embedFile :: (FilePath, MimeType) -> IO EmbeddableEntry
embedFile (file, mime) = do
    f <- BL.readFile $ "static" </> file
    return $ EmbeddableEntry {
            eLocation = T.pack file
        ,   eMimeType = mime
        ,   eContent  = Left (hash f, f)
        }

mkEmbedded :: IO [EmbeddableEntry]
mkEmbedded = mapM embedFile staticFiles
```

and i create my application like this:

``` haskell
{-# LANGUAGE TemplateHaskell, OverloadedStrings #-}
module Main where

import Static (mkEmbedded)
import Network.Wai (Application)
import Network.Wai.Application.Static
import WaiAppStatic.Storage.Embedded
import Network.Wai.Handler.Warp (run)
import WaiAppStatic.Types (unsafeToPiece)

myApp :: Application
myApp = staticApp $(mkSettings mkEmbedded) {ssIndices = [unsafeToPiece "index.html"]}

main :: IO ()
main = run 3000 myApp

```

i also tried `ssRedirectToIndex = True`, but directly visit `localhost:3000` result in 404, `localhost:3000/index.html`worked as expected, is there something wrong with my code?

**Comment by snoyberg** [comment]
Looking at the code, it appears that directory listings and indices are features that are simply not supported currently. Such support could be added in theory, but it would require changes to the internal structure. I don't have bandwidth to make those changes myself right now, but I'd be happy to provide guidance on them (I don't think they're that difficult, but will take some time to get everything in shape). If you're interested in doing so, that would be great. Otherwise, I'll provide a doc fix explaining that this use case is not currently supported.

Thank you for reporting.

**Comment by winterland1989** [comment]
As a new haskell beginner(just pick it up 2 month ago)... I'll try what i can do, thanks!

**Comment by soenkehahn** [comment]
@snoyberg: I just ran into the same issue. Did you get around to provide the doc fix and I missed that part?

**Comment by snoyberg** [comment]
No, I didn't. Would you be able to send a PR?

**Comment by soenkehahn** [comment]
I created a (failing) test-case that demonstrates the behavior, see #460. I haven't looked into the code yet, but is there any reason `defaultWebAppSettings` shouldn't support `ssIndices`? If there are no objections I'll try to make this work.

**Comment by snoyberg** [comment]
None that I'm aware of.

**Comment by Vlix** [comment]
@soenkehahn A bit late, but do you remember if this was fixed? #460 seems to do what this issue was opened for, if I understand it correctly? 🤔

**Comment by soenkehahn** [comment]
I don't remember the details at all, but I did claim in #460 that it fixes this issue. So I think this can be closed.
